### PR TITLE
Adding drainBeforeDelete field to enable the feature in provisioning tests

### PR DESCRIPTION
**Pull request by thaneunsoo**
This issue is a forward port of [42839](https://github.com/rancher/rancher/pull/42839)

## Issue: 
 
[513](https://github.com/rancher/qa-tasks/issues/513)
 
## Problem

We don't have the option to set the value for drainBeforeDelete in our automation suite. We want to include test cases for this feature and need this option in our framework.
 
## Solution

We are adding the drainBeforeDelete field to allow us to use this feature in our automation suite.
 
### Manual Testing

Manual testing was done on local setup to ensure that the field was being added correctly.

### Automated Testing

Jenkins jobs will be available upon request

## QA Testing Considerations

This is for the qa team to be able to run automated tests around the drainBeforeDelete feature
 
### Regressions Considerations

There are no regressions as the feature already exists and we are just adding the field to enable the feature on the automation suite.

### CODE REVIEW

File: tests/framework/extensions/clusters/clusters.go [hunk 1]
```diff
@@ -396,6 +396,12 @@ func NewK3SRKE2ClusterConfig(clusterName, namespace string, clustersConfig *Clus
 		spec.DefaultPodSecurityAdmissionConfigurationTemplateName = clustersConfig.PSACT
 	}
 
+	if clustersConfig.DrainBeforeDelete {
+		for i := range rkeConfig.MachinePools {
+			rkeConfig.MachinePools[i].DrainBeforeDelete = clustersConfig.DrainBeforeDelete
+		}
+	}
+
```
Comment (slickwarren) [inline tests/framework/extensions/clusters/clusters.go:404]: we should move this up above where 'spec' is defined

**Comment by igomez06** [comment]
fix the merge conflicts

**Review by igomez06 — APPROVED** [review]
LGTM

**Review by markusewalker — APPROVED** [review]
LGTM

**Review by slickwarren — APPROVED** [review]

**Pull request merged**